operations/program/program.go

```diff
@@ -0,0 +1,49 @@
+/*
+ Copyright 2016 Padduck, LLC
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ 	http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+package program
+
+import (
+	"fmt"
+	"github.com/pufferpanel/pufferpanel/v2"
+	"github.com/pufferpanel/pufferpanel/v2/programs"
+)
+
+type Program struct {
+	Action  string
+	Program *programs.Program
+}
+
+func (d Program) Run(env pufferpanel.Environment) error {
+	p := d.Program
+	switch d.Action {
+	case "install":
+		return p.Install()
+	case "start":
+		return p.Start()
+	case "stop":
+		return p.Stop()
```

Comment:
This needs a wait, Stop doesn't wait for it to end, so you'd have to add a check for that here